src/components/gltf-model-plus.js

```diff
@@ -341,127 +340,221 @@ function runMigration(version, json) {
   }
 }
 
-const loadLightmap = async (parser, materialIndex) => {
-  const lightmapDef = parser.json.materials[materialIndex].extensions.MOZ_lightmap;
-  const [material, lightMap] = await Promise.all([
-    parser.getDependency("material", materialIndex),
-    parser.getDependency("texture", lightmapDef.index)
-  ]);
-  material.lightMap = lightMap;
-  material.lightMapIntensity = lightmapDef.intensity !== undefined ? lightmapDef.intensity : 1;
-  return lightMap;
-};
-
 let ktxLoader;
 
-export async function loadGLTF(src, contentType, onProgress, jsonPreprocessor) {
-  let gltfUrl = src;
-  let fileMap;
-
-  if (contentType && (contentType.includes("model/gltf+zip") || contentType.includes("application/x-zip-compressed"))) {
-    fileMap = await extractZipFile(gltfUrl);
-    gltfUrl = fileMap["scene.gtlf"];
+class GLTFHubsPlugin {
+  constructor(parser, jsonPreprocessor, fileMap) {
+    this.parser = parser;
+    this.jsonPreprocessor = jsonPreprocessor;
+    this.fileMap = fileMap;
+
+    // The latest GLTFLoader doesn't use ImageBitmapLoader for Firefox
+    // because the latest ImageBitmapLoader passes an option parameter
+    // to createImageBitmap() but Firefox createImageBitmap() fails
+    // if an option parameter is passed (known bug and already reported to bugzilla).
+    // But our r111 based ImageBitmapLoader doesn't pass the option parameter yet
+    // so we can use ImageBitmapLoader even for Firefox.
+    // When we replace our Three.js fork with the latest official one
+    // we need to revisit this workaround.
+    if (!parser.textureLoader.isImageBitmapLoader && typeof createImageBitmap !== undefined) {
+      parser.textureLoader = new THREE.ImageBitmapLoader(parser.options.manager);
```

Comment:
ImageBitmapLoader returns an ImageBitmap not a Texture instance I thought. Doesn't this need to be a some kind of texture loader?

Reply:
[`GLTFLoader` makes `CanvasTexture` from `ImageBitmap` if `textureLoader` is `ImageBitmapLoader`](https://github.com/mrdoob/three.js/blob/r128/examples/jsm/loaders/GLTFLoader.js#L2578-L2586) so this line should be fine.

But this line may be confusing for people who don't know the inside of `GLTFLoader`. I would add comment.